homeassistant/components/zwave/climate.py

```diff
@@ -131,21 +142,38 @@ def update_properties(self):
         if self.values.mode:
             self._hvac_list = []
             self._hvac_mapping = {}
-            hvac_list = self.values.mode.data_items
-            if hvac_list:
-                for mode in hvac_list:
-                    ha_mode = HVAC_STATE_MAPPINGS.get(mode)
+            self._preset_list = []
+            self._preset_mapping = {}
+
+            mode_list = self.values.mode.data_items
+            if mode_list:
+                for mode in mode_list:
+                    ha_mode = HVAC_STATE_MAPPINGS.get(
+                        mode.lower() if isinstance(mode, str) else mode)
+                    ha_preset = PRESET_MAPPINGS.get(
+                        mode.lower() if isinstance(mode, str) else mode)
```

Comment:
What is the issue they you try to solve with check the type?

Reply:
mode.lower() works only for string types. Errors occur if the lower() method is called on sth other than strings. There were some tests that uses ints to specify modes, so I've decided to test this.

Reply:
Let cast it before to a string

Reply:
Done